src/main/scala/is/hail/methods/KinshipMatrix.scala

```diff
@@ -2,29 +2,30 @@ package is.hail.methods
 
 import java.io.DataOutputStream
 
-import breeze.linalg.SparseVector
 import is.hail.HailContext
 import is.hail.annotations.Annotation
 import is.hail.expr.types._
 import is.hail.linalg.BlockMatrix
+import is.hail.table.Table
 import is.hail.utils._
 import org.apache.spark.mllib.linalg.Vectors
 import org.apache.spark.mllib.linalg.distributed.{IndexedRow, IndexedRowMatrix}
 
 import scala.collection.Searching._
-import scala.collection.JavaConverters._
 
 /**
   * Represents a KinshipMatrix. Entry (i, j) encodes the relatedness of the ith and jth samples in sampleIds.
   */
 object KinshipMatrix {
-  def apply(hc: HailContext, sampleSignature: Type, matrix: BlockMatrix, sampleIds: java.util.ArrayList[Annotation], numVariantsUsed: java.lang.Long): KinshipMatrix = {
-    KinshipMatrix(hc, sampleSignature, matrix.toIndexedRowMatrix(), sampleIds.asScala.toArray, numVariantsUsed.longValue())
+  def apply(hc: HailContext, matrix: BlockMatrix, colKeys: Table, numVariantsUsed: java.lang.Long): KinshipMatrix = {
+    require(colKeys.signature.size == 1 && colKeys.signature.fields(0).typ.isOfType(TString()))
+    val sampleIds: Array[Annotation] = colKeys.collect().map(_.getAs[Annotation](0))
+    
+    KinshipMatrix(hc, TString(), matrix.toIndexedRowMatrix(), sampleIds, numVariantsUsed.longValue())
```

Comment:
why does this still have a string requirement?

Reply:
can't it be the struct now?

Reply:
See comment. This class is going away.